datasets/ami/ami.py

```diff
@@ -0,0 +1,581 @@
+                        words.append(text)
+                        word_speakers.append(speaker)
+                    else:
+                        logger.warning(
+                            f"Annotation {word_id} of file {path} is missing information about"
+                            "either word_start_time or word_end_time. Skipping sample..."
+                        )
+
+        return AMI._sort(word_start_times, [word_ids, word_start_times, word_end_times, words, word_speakers])
+
+    @staticmethod
+    def _extract_segments_annotations(paths):
+        segment_ids = []
+        channels = []
+        segment_start_times = []
+        segment_end_times = []
+        segment_speakers = []
+
+        for path in paths:
+            speaker = path.split(".")[-3]
+
+            with open(path, "r", encoding="utf-8") as segments_file:
+                root = ET.parse(segments_file).getroot()
+                for type_tag in root.findall("segment"):
+                    segment_ids.append(type_tag.get("{http://nite.sourceforge.net/}id"))
+                    segment_start_times.append(float(type_tag.get("transcriber_start")))
+                    segment_end_times.append(float(type_tag.get("transcriber_end")))
+                    channels.append(type_tag.get("channel"))
+                    segment_speakers.append(speaker)
+
+        return AMI._sort(
+            segment_start_times, [segment_ids, segment_start_times, segment_end_times, channels, segment_speakers]
+        )
+
+    def _generate_examples(self, annotation_path, samples_paths, ids, verification_ids):
+        logger.info(f"⏳ Generating {', '.join(ids)}")
+
+        # number of audio files of config
+        num_audios = len(self.config.dl_path_formats)
+
+        # filter missing ids
+        ids = list(filter(lambda n: n not in self.config.missing_files, ids))
+
+        # audio
+        samples_paths_dict = {}
+        for i, _id in enumerate(ids):
+            sample_paths = samples_paths[num_audios * i : num_audios * (i + 1)]
+            sample_verification_id = set(verification_ids[num_audios * i : num_audios * (i + 1)])
+
+            # make sure that multi microphone samples are correctly atttributed to labels
+            if len(sample_verification_id) > 1 or next(iter(sample_verification_id)) != _id:
+                raise ValueError(
+                    f"Incorrect dataset generation. The files {sample_paths} of id {_id} have incorrect verification_ids {sample_verification_id}."
+                )
+
+            # set correct files correctly
+            samples_paths_dict[_id] = sample_paths
+
+        # words
+        words_paths = {
+            _id: [os.path.join(annotation_path, "words/{}.{}.words.xml".format(_id, speaker)) for speaker in _SPEAKERS]
+            for _id in ids
+        }
+        words_paths = {_id: list(filter(lambda path: os.path.isfile(path), words_paths[_id])) for _id in ids}
```

Comment:
FYI `os.path.isfile` is not supported yet for dataset streaming, but we can add it I think :)

(we just need to check in the remote zip file if the file exists, instead of checking locally)